### PR TITLE
Issue #3193302 by tBKoT: Add books and pages to select list of custom content list block

### PR DESCRIPTION
**Pull request by dmytriikaiun**
## Problem
Currently, when I create a list of content in Dashboard (probably Landin page too) I can only select Topics, Events, Groups, and Discussions.
It would be great if also book pages and basic pages were included.

## Solution
Create submodules with plugins to provide the possibility to select Books and Basic pages.

## Issue tracker
https://getopensocial.atlassian.net/browse/YANG-4390
https://www.drupal.org/project/social/issues/3193302

## How to test
- [x] Create dashboard
- [x] Add the `Custom content list block`
- [x] As `Type of content` you should have the possibility to select the `Book page` or `Basic page`

## Screenshots
N/A

## Release notes
Now admins have the possibility to select `Book page` or `Basic page` in the `Custom content list block `

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Review by ribel — APPROVED** [review]
I have tested it and all works as expected.

**Comment by ribel** [comment]
@dmytriikaiun Please resolve conflicts here when you have time.

**Comment by ribel** [comment]
@dmytriikaiun Please merge and change base branch to 10.1.x and fix Coding Standards found in Travis.

**Comment by ribel** [comment]
@ronaldtebrake We added test coverage for Content List Block within this PR, so it's ready to merge when Travis will be fixed.

File: modules/social_features/social_book/config/optional/core.entity_view_display.node.book.small_teaser.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    - field.field.node.book.body
+    - field.field.node.book.field_book_comments
+    - field.field.node.book.field_book_image
+    - field.field.node.book.field_content_visibility
+    - field.field.node.book.field_files
+    - image.style.social_medium
+    - node.type.book
+  module:
+    - image
+    - layout_builder
```
Comment (ronaldtebrake) [inline modules/social_features/social_book/config/optional/core.entity_view_display.node.book.small_teaser.yml:15]: are all these dependencies really needed? 🤔

File: modules/social_features/social_book/social_book.install [hunk 2]
```diff
@@ -241,3 +242,17 @@ function social_book_update_8902() {
   // Make sure we clear cached definitions for the fields.
   \Drupal::service('entity_field.manager')->clearCachedFieldDefinitions();
 }
+
+/**
+ * Add small_teaser view mode for book.
+ */
+function social_book_update_8903() {
+  $config_path = drupal_get_path('module', 'social_book') . '/config/optional';
```
Comment (ronaldtebrake) [inline modules/social_features/social_book/social_book.install:250]: Please use a static file for this, config/optional can change when we export in the future and than our updates might fail.

File: modules/social_features/social_page/social_page.install [hunk 3]
```diff
@@ -191,3 +192,17 @@ function social_page_update_8907() {
     }
   }
 }
+
+/**
+ * Add small_teaser view mode for page.
+ */
+function social_page_update_8908() {
+  $config_path = drupal_get_path('module', 'social_page') . '/config/install';
```
Comment (ronaldtebrake) [inline modules/social_features/social_page/social_page.install:200]: can we use a static file here as well?

**Comment by ronaldtebrake** [comment]
Looks good to me other than the minor remarks :)

File: modules/social_features/social_page/social_page.install (hunk 3 above)
Comment (ribel) [reply modules/social_features/social_page/social_page.install]: I see it was fixed.

File: modules/social_features/social_book/social_book.install (hunk 2 above)
Comment (ribel) [reply modules/social_features/social_book/social_book.install]: I see it was fixed.

File: modules/social_features/social_book/config/optional/core.entity_view_display.node.book.small_teaser.yml (hunk 1 above)
Comment (ribel) [reply modules/social_features/social_book/config/optional/core.entity_view_display.node.book.small_teaser.yml]: I see it was fixed.

**Pull request merged**